### DISCUSSION TITLE
M600 Not enough space for the purge

### DISCUSSION THREAD

**Discussion by Drakkan69** [Guilouz/Creality-Helper-Script-Wiki#492] [Issues]
English message below :
Je ne sais pas si vous le savait, mais la commande M600 purge très près du lit si vous changé de filament à l’intérieur de 48 mm. Il n’y a pas assez d’espace entre le lit et la buse. Ceci survient parce qu’une partie du code de la macro RESUME sera exécuté à la suite de la commande M600. Je vous laisse trouver la solution idéale puisque vos connaissances son beaucoup supérieur au mienne. Moi j’ai fait une solution rapide j’ai mis le code suivant à la ligne 34:

**Code ajouter :**
SET_GCODE_VARIABLE MACRO=PRINTER_PARAM VARIABLE=z_safe_pause VALUE=0

**Partie de la macro RESUME qui sera exécuté :**
{% if z_resume_move > 2 %}
    {% set z_resume_move = z_resume_move - 2 %}
    G91
    G1 Z-{z_resume_move} F600
    M400
  {% endif %}

P.S. J’ai mis à jour le script et le M600, la version 5.1.0.1 et la version 5.1.0.3 ont le même problème. Le problème de la 5.1.01 était identique mais ne survenait pas dans les mêmes conditions.

Google translate :
I don't know if you knew this, but the M600 control purge very close to the bed if you changed the filament within 48mm. There is not enough space between the bed and the nozzle. This occurs because part of the RESUME macro code will be executed following the M600 command. I'll let you find the ideal solution since your knowledge is much greater than mine. I did a quick solution, I put the following code on line 34:

**Add code line 34:**
SET_GCODE_VARIABLE MACRO=PRINTER_PARAM VARIABLE=z_safe_pause VALUE=0

**Part of the RESUME macro that will be executed:**
{% if z_resume_move > 2 %}
    {% set z_resume_move = z_resume_move - 2 %}
    G91
    G1 Z-{z_resume_move} F600
    M400
  {% endif %}

P.S. I updated the script and M600, version 5.1.0.1 and version 5.1.0.3 have the same problem. The problem in 5.1.0.1 was identical but did not occur under the same conditions.

Thanks and continue your excellent job.

**Comment by BloodSteyn** [comment]
I have just experienced the low purge. Made a huge blob directly under the nozzle. It is working, but I had to manually remove the blob.

I can't edit the M600 Macro file though.
